### PR TITLE
web: redesign repository settings page.

### PR DESCRIPTION
**Pull request by sashaostrikov**
It includes adding headings for "Code host" and "Repository name" sections, making code host list less heavy (stripping code host card borders and icon) and removing unnecessary form for repository name, substituting it with a CopyableText component.

## Before
image

### Single code host
image

### Multiple code hosts
image

Test plan:
Local sg run and visual checks.

Part of https://github.com/sourcegraph/sourcegraph/issues/40504

## App preview:

- [Web](https://sg-web-ao-ui-repo-settings-code-host.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.01% (+1.84 kb) | 0.02% (+1.84 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ao/ui/repo-settings-code-host/compare-report.html;tab=live_object) for a full comparison between the commits f545748 and 4d27d54 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/components/externalServices/ExternalServiceCard.tsx [hunk 1]
```diff
@@ -11,44 +11,57 @@ import styles from './ExternalServiceCard.module.scss'
...
     let cardTitle = <H3 className={shortDescription ? 'mb-0' : 'mt-1 mb-0'}>{title}</H3>
     cardTitle = tooltip ? <Tooltip content={tooltip}>{cardTitle}</Tooltip> : cardTitle
     const children = (
-        <div className={classNames('p-3 d-flex align-items-start border' + (enabled ? '' : ' text-muted'), className)}>
+        <div
+            className={classNames(
+                'd-flex align-items-start' + (bordered ? ' p-3 border' : ' p-1') + (enabled ? '' : ' text-muted'),
+                className
+            )}
+        >
```
Comment (indradhanush) [inline client/web/src/components/externalServices/ExternalServiceCard.tsx:64]: Usually the div definition is one line. I think it's okay to have it in the same line.

**Review by indradhanush — APPROVED** [review]
LGTM. One minor comment that I'll upto you.

File: client/web/src/components/externalServices/ExternalServiceCard.tsx (hunk 1 above)
Comment (sashaostrikov) [reply client/web/src/components/externalServices/ExternalServiceCard.tsx:64]: it's how prettier does it, we have some multi-line divs in our code

**Pull request merged**